pkg/broker/broker.go

```diff
@@ -265,23 +275,35 @@ func (a AnsibleBroker) Deprovision(instanceUUID uuid.UUID) (*DeprovisionResponse
 	//    decide what's important?
 	//    * if noerror: delete serviceInstance entry with Dao
 	////////////////////////////////////////////////////////////
-	var err error
-	var instance *apb.ServiceInstance
-	instanceId := instanceUUID.String()
+	instanceID := instanceUUID.String()
 
-	if err = a.validateDeprovision(instanceId); err != nil {
+	if err := a.validateDeprovision(instanceID); err != nil {
 		return nil, err
 	}
-
-	if instance, err = a.dao.GetServiceInstance(instanceId); err != nil {
+	instance, err := a.dao.GetServiceInstance(instanceID)
+	/// Handle case where service is not found in etcd
+	if client.IsKeyNotFound(err) {
+		a.log.Debug("unable to find service instance - %#v", err)
+		return nil, ErrorNotFound
+	}
+	// bubble up  error
+	if err != nil {
+		a.log.Error("unknown etcd error - %#v", err)
```

Comment:
Even though we aren't catching the error explicitly, I think etcd will know the error.  Maybe it should be more like ```Error from etcd - ...```.